RePak/src/RePak.cpp

```diff
@@ -119,6 +120,41 @@ int main(int argc, char** argv)
 
     doc.ParseStream(isw);
 
+    if (doc.HasParseError()) {
+        int lineNum = 1;
+        int columnNum = 0;
+        std::string lastLine = "";
+        std::string curLine = "";
+
+        int offset = doc.GetErrorOffset();
+        ifs.clear();
+        ifs.seekg(0, std::ios::beg);
+        IStreamWrapper isw{ ifs };
+
+        for (int i = 0; ; i++)
+        {
+            char c = isw.Take();
+            curLine.push_back(c);
+            if (c == '\n')
+            {
+                if (i >= offset)
+                    break;
+                lastLine = curLine;
+                curLine = "";
+                lineNum++;
+                columnNum = 0;
+            }
+            else
+            {
+                if (i < offset)
+                    columnNum++;
+            }
+        }
+
+        // this could probably be formatted nicer
+        Error("Failed to parse map file: \n\nLine %i, Column %i\n%s\n\n%s%s%s\n", lineNum, columnNum, GetParseError_En(doc.GetParseError()), lastLine.c_str(), curLine.c_str(), (std::string(columnNum, ' ') += '^').c_str());
```

Comment:
```suggestion
        // this could probably be formatted nicer
        Error("Failed to parse map file: \n\nLine %i, Column %i\n%s\n\n%s%s%s\n", 
            lineNum, 
            columnNum, 
            GetParseError_En(doc.GetParseError()), 
            lastLine.c_str(), 
            curLine.c_str(), 
            (std::string(columnNum, ' ') += '^').c_str());
```
Do we prefer this formatting or existing formatting?